scaffolder-templates/nodejs-backend-template/github-action-skeleton/Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Stage 1 - Install dependencies
+FROM registry.access.redhat.com/ubi9/nodejs-18-minimal:latest AS deps
+
+COPY package.json package-lock.json ./
+RUN npm ci
+
+# Stage 3 - Production image, copy all the files and start NodeJS
+FROM registry.access.redhat.com/ubi9/nodejs-18-minimal:latest AS runner
+
+ENV NODE_ENV production
+
+# Switch to nodejs user
+USER 1001
+
+COPY . .
+COPY --from=deps /opt/app-root/src .
+
+EXPOSE 3000
+ENV PORT 3000
+
+# The fix-permissions script is important when operating in environments that dynamically use a random UID at runtime, such as OpenShift.
+RUN fix-permissions ./
+
+CMD ["node", "src/index.ts"]
```

Comment:
We can do more testing, but it seems to be working fine

Reply:
Ah, from digging in, it looks like it is working here as the example code in `scaffolder-templates/nodejs-backend-template/github-action-skeleton/src/index.ts` contains plain JavaScript and no TypeScript syntax - so Node.js is treating like a plain JS file and executing it fine. But, adding any TypeScript syntax like `type Hello = "World" | "Universe";` and it'll not execute. 

This might be confusing to users, as if they're using TypeScript they'll need to do the `tsc` transpile step and run the output JS files.

Reply:
Thanks for the information. We will improve this to avoid any confusion to users.

Reply:
I think an option would be to assume plain JavaScript by default and have any sample code in plain `.js` files. Or, implement a full TypeScript setup with a `tsc` build/transpile flow in the `Dockerfile`. But, perhaps, having both examples/templates would be worthwhile? 

Let me know if I can help at all. Thanks!